flow/dynamicconf/dynamicconf.go

```diff
@@ -0,0 +1,69 @@
+package dynamicconf
+
+import (
+	"context"
+	"log/slog"
+	"strconv"
+	"time"
+
+	utils "github.com/PeerDB-io/peer-flow/connectors/utils/catalog"
+	"github.com/jackc/pgx/v5/pgtype"
+	"github.com/jackc/pgx/v5/pgxpool"
+)
+
+func dynamicConfKeyExists(ctx context.Context, conn *pgxpool.Pool, key string) bool {
+	var exists pgtype.Bool
+	query := "SELECT EXISTS(SELECT 1 FROM alerting_settings WHERE config_name = $1)"
+	err := conn.QueryRow(ctx, query, key).Scan(&exists)
+	if err != nil {
+		slog.Error("Failed to check if key exists: %v", err)
+		return false
+	}
+
+	return exists.Bool
+}
+
+func dynamicConfUint32(ctx context.Context, key string, defaultValue uint32) uint32 {
+	conn, err := utils.GetCatalogConnectionPoolFromEnv()
```

Comment:
how often do we expect this to be called across all variables? we might either want to open and close a single connection or open a single pool and cache it.

Also pool isn't being closed

Reply:
this is a global pool, it isn't closed.